stellargraph/mapper/sequences.py

```diff
@@ -527,3 +530,128 @@ def __len__(self):
+        degree_matrix.setdiag(degree_matrix_diag)
+        degree_matrix = degree_matrix.tocsr()
+        adj = degree_matrix @ adj @ degree_matrix
+        return adj
+
+    def __getitem__(self, index):
+        graphs = self.graphs[
+            index * self.batch_size : (index * self.batch_size) + self.batch_size
+        ]
+        adj_graphs = [graph.to_adjacency_matrix() for graph in graphs]
+
+        # The number of nodes for the largest graph in the batch. We are going to pad with 0 rows and columns
+        # the adjacency and node feature matrices (only the rows in this case) to equal in size the adjacency and
+        # feature matrices of the largest graph.
+        max_nodes = max([graph.number_of_nodes() for graph in graphs])
+
+        # The operations to normalize the adjacency matrix are a bit slow.
+        # FIXME: Either optimize this or implement as a layer.
+        if self.normalize_adj:
+            adj_graphs = [self.__normalize_adj(adj).toarray() for adj in adj_graphs]
+
+        graph_targets = None
+        #
+        if self.targets is not None:
+            graph_targets = self.targets[
+                index * self.batch_size : (index * self.batch_size) + self.batch_size
+            ]
+
+        # pad adjacency and feature matrices to equal the size of those from the largest graph
+        features = [
+            np.pad(
+                graph.node_features(graph.nodes()),
+                pad_width=(0, max_nodes - graph.number_of_nodes()),
+                mode="constant",
+                constant_values=0.0,
+            )[:, : self.node_features_size]
+            for graph in graphs
+        ]
+        features = np.stack(features)
+        adj_graphs = [
+            np.pad(
+                adj_graph,
+                pad_width=(0, max_nodes - len(adj_graph)),
+                mode="constant",
+                constant_values=0.0,
+            )
+            for adj_graph in adj_graphs
+        ]
+        adj_graphs = np.stack(adj_graphs)
+        masks = np.full((len(graphs), max_nodes), fill_value=False, dtype=np.bool)
+        for index, graph in enumerate(graphs):
+            masks[index, : graph.number_of_nodes()] = True
+
+        # features is array of dimensionality
+        #      batch size x max number of nodes in graph x node feature dimensionality
+        # masks is array of dimensionality
+        #      batch size x max number of nodes
+        # adj_graphs is array of dimensionality
+        #      batch size x max number of nodes x max number of nodes
+        # graph_targets is array of dimensionality
+        #      batch size x number of classes
+        return [features, masks, adj_graphs], graph_targets
+
+    def on_epoch_end(self):
+        """
+         Shuffle all graphs at the end of each epoch
+        """
+        indexes = list(range(len(self.graphs)))
+        random.shuffle(indexes)
+        self.graphs = [self.graphs[i] for i in indexes]
```

Comment:
If you wanted to, `graphs` could be stored as a numpy array (e.g. `self.graphs = np.asanyarray(graphs)` in `__init__`) which will have `dtype=object`. This then allows this code to be:

```python
indexes = np.random.permutation(len(self.graphs))
self.graphs = self.graphs[indexes]
self.targets = self.targets[indexes]
```

Reply:
I have updated as suggested.